### PR TITLE
Update kbd, add tests

### PR DESCRIPTION
**Pull request by davidak**
###### Motivation for this change

kbd was outdated and kind of abandoned, held together with patches

the new version includes neo and dvorak-programmer (which where previously added as extra packages) and more

i don't have the hardware to test it on armv5tel-linux and mips

this package seem to be very important. 11503 packages have it as dependency, including systemd

i added myself as maintainer, but i would like to have a co-maintainer who migh know a bit more about the domain

i updated the test and added it to the package

cc @edolstra @aszlig @wkennington @Profpatsch @abbradar @rnhmjoj 

please review carefully!

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [x] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests)) ( `nix-build . -A kbd.tests` )
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [X] Ensured that relevant documentation is up to date
- [X] manual build and checked
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by aszlig** [comment]
@ofborg test keymap

File: nixos/tests/kbd.nix [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    assert "${pkgs.kbd.version}" in machine.succeed("loadkeys -V")
+    # test if keymaps are installed
+    machine.succeed("loadkeys us")
+    machine.succeed("loadkeys fr")
+    machine.succeed("loadkeys de")
+    machine.succeed("loadkeys neo")
+    machine.succeed("loadkeys bone")
+    machine.succeed("loadkeys es-olpc")
+    machine.succeed("loadkeys dvorak")
+    machine.succeed("loadkeys colemak")
```
Comment (aszlig) [inline nixos/tests/kbd.nix:18]: As mentioned in my comment to @ofborg, we already have a [`keymap` test](https://github.com/NixOS/nixpkgs/blob/4b21d06df01d181ad91f5ba224b9cd3819f4949a/nixos/tests/keymap.nix) which not only tests `loadkeys` but tests whether the mapping is correctly applied. At the moment only `azerty`, `colemak`, `dvorak`, `dvp`, `neo` and  `qwertz` is tested, so if you're using another layout, feel free to add a subtest there.

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 119096` at 87c37062 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>782 packages marked as broken and skipped:</summary>
  <ul>
    <li>adobe-reader</li>
    <li>adoptopenjdk-hotspot-bin-13</li>
    <li>adoptopenjdk-hotspot-bin-14</li>
    <li>adoptopenjdk-jre-hotspot-bin-13</li>
    <li>adoptopenjdk-jre-hotspot-bin-14</li>
    <li>adoptopenjdk-jre-openj9-bin-13</li>
    <li>adoptopenjdk-jre-openj9-bin-14</li>
    <li>adoptopenjdk-openj9-bin-13</li>
    <li>adoptopenjdk-openj9-bin-14</li>
    <li>agdaPackages.iowa-stdlib</li>
    <li>...</li>
  </ul>
</details>
<details>
  <summary>10291 packages skipped due to time constraints:</summary>
  <ul>
    <li>AusweisApp2</li>
    <li>DisnixWebService</li>
    <li>EBTKS</li>
    <li>EmptyEpsilon</li>
    <li>Fabric (python38Packages.Fabric)</li>
    <li>MIDIVisualizer</li>
    <li>MMA</li>
    <li>OSCAR</li>
    <li>OVMF</li>
    <li>OVMF-CSM</li>
    <li>...</li>
  </ul>
</details>
<details>
  <summary>1 package built successfully:</summary>
  <ul>
    <li>kbd</li>
  </ul>
</details>
<details>
  <summary>1 suggestion:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/maintainers-missing.md">maintainers-missing</a>

Package does not have a maintainer. Consider adding yourself?

Near pkgs/os-specific/linux/kbd/default.nix:53:5:
```
   |
53 |     maintainers = with maintainers; [  ];
   |     ^
```
</li>
  </ul>
</details>

File: nixos/tests/kbd.nix (hunk 1 above)
Comment (davidak) [reply nixos/tests/kbd.nix:18]: i haven't found that. thanks!

**Comment by aszlig** [comment]
Cc: @viric, who originally disabled `-Werror` for ARMv5 (2dfeeb47bb8581fa166c9db3d054470e63390f3a) and MIPS (b368b4b199aa81207f9804aa0783d9a03e0356f8).

**Comment by davidak** [comment]
`-Werror` is not in `src/Makefile.am` anymore

**Comment by davidak** [comment]
@ofborg test kbd.tests

File: pkgs/top-level/all-packages.nix [hunk 2]
```diff
@@ -19593,8 +19593,6 @@ in
 
   kbd = callPackage ../os-specific/linux/kbd { };
 
-  kbdKeymaps = callPackage ../os-specific/linux/kbd/keymaps.nix { };
```
Comment (SuperSandro2000) [inline pkgs/top-level/all-packages.nix:19596]: Please add an alias to aliases.nix.

File: pkgs/os-specific/linux/kbd/default.nix [hunk 3]
```diff
@@ -1,15 +1,15 @@
-{ lib, stdenv, fetchurl, autoreconfHook,
-  gzip, bzip2, pkg-config, flex, check,
+{ lib, stdenv, fetchurl, nixosTests, autoreconfHook,
+  pkg-config, flex, check,
   pam, coreutils
 }:
```
Comment (SuperSandro2000) [inline pkgs/os-specific/linux/kbd/default.nix:4]:
```suggestion
{ lib, stdenv, fetchurl, nixosTests, autoreconfHook
, pkg-config, flex, check, pam, coreutils
}:
```

File: nixos/modules/config/console.nix [hunk 4]
```diff
@@ -82,8 +82,7 @@ in
 
     packages = mkOption {
       type = types.listOf types.package;
-      default = with pkgs.kbdKeymaps; [ dvp neo ];
-      defaultText = "with pkgs.kbdKeymaps; [ dvp neo ]";
```
Comment (SuperSandro2000) [inline nixos/modules/config/console.nix:86]: Why did you remove them?

File: pkgs/top-level/all-packages.nix (hunk 2 above)
Comment (davidak) [reply pkgs/top-level/all-packages.nix:19596]: should it duplicate the information in the release notes or link to it?

File: nixos/modules/config/console.nix (hunk 4 above)
Comment (davidak) [reply nixos/modules/config/console.nix:86]: as written in the pr description, they are included in the new version of kbd

you can see that the tests still works

**Comment by davidak** [comment]
The package and tests build successful, see https://github.com/NixOS/nixpkgs/pull/119096/checks?check_run_id=2316173120

File: nixos/tests/keymap.nix [hunk 5]
```diff
@@ -142,6 +161,7 @@ in pkgs.lib.mapAttrs mkKeyboardTest {
       symbols.expect = [ "&" "[" "{" "}" "(" "=" "*" ")" "+" "]" "!" ];
     };
 
+    extraConfig.console.keyMap = "dvorak-programmer";
     extraConfig.services.xserver.layout = "us";
     extraConfig.services.xserver.xkbVariant = "dvp";
```
Comment (davidak) [inline nixos/tests/keymap.nix:166]: I don't understand why i have do use `dvp` in one place and `dvorak-programmer` in the other. Where does dvp come from?

The actual file is called https://github.com/legionus/kbd/blob/master/data/keymaps/i386/dvorak/dvorak-programmer.map and in the whole repo is no mention of dvp.

Comment (apfelkuchen6) [reply nixos/tests/keymap.nix:166]: dvp is the name of the layout in [xkeyboard-config](https://github.com/freedesktop/xkeyboard-config/blob/master/symbols/us#L555) and thus has to be used to configure the layout of the xorg display server. Note that the test checks the keyboard layouts for both the linux console and the xorg server.

Comment (davidak) [reply nixos/tests/keymap.nix:166]: thanks. i clarified it in the documentation

File: pkgs/os-specific/linux/kbd/default.nix [hunk 6]
```diff
@@ -57,16 +39,16 @@ stdenv.mkDerivation rec {
     done
   '';
 
-
   buildInputs = [ check pam ];
   nativeBuildInputs = [ autoreconfHook pkg-config flex ];
 
-  makeFlags = [ "setowner=" ];
```
Comment (davidak) [inline pkgs/os-specific/linux/kbd/default.nix:64]: I just removed the line `makeFlags = [ "setowner=" ];` because it seem to work without it. Is there any reason why it might be needed?

cc @abbradar who added it in https://github.com/NixOS/nixpkgs/commit/853018124668d83bc17815da52d0e68b1a163eb7

That's the last bit where i'm unsure, the rest of the PR seems good to me now.

File: pkgs/os-specific/linux/kbd/default.nix [hunk 7]
```diff
@@ -1,15 +1,14 @@
-{ lib, stdenv, fetchurl, autoreconfHook,
-  gzip, bzip2, pkg-config, flex, check,
-  pam, coreutils
+{ lib, stdenv, fetchurl, nixosTests, autoreconfHook,
+  pkg-config, flex, check, pam, coreutils
```
Comment (aszlig) [inline pkgs/os-specific/linux/kbd/default.nix:2]: I don't think we have specified this in the documentation, but I believe the inofficially™ agreed upon style is more like this:

```suggestion
{ lib, stdenv, fetchurl, nixosTests, autoreconfHook
, pkg-config, flex, check, pam, coreutils
```

File: pkgs/os-specific/linux/kbd/default.nix (hunk 6 above)
Comment (aszlig) [reply pkgs/os-specific/linux/kbd/default.nix:64]: It wasn't @abbradar but @edolstra in 65e004b75775fa4e3d5f9a40be92e0655774b005.

Nevertheless, there is no ocurrence of `setowner` anywhere in the source code, so I think it's fine to drop this.

File: pkgs/os-specific/linux/kbd/default.nix (hunk 7 above)
Comment (davidak) [reply pkgs/os-specific/linux/kbd/default.nix]: i think the closest we have to a code style convention is nixpkgs-fmt , so i used it (for the first time) to format it

**Comment by davidak** [comment]
I tried to rebuild my system with this, but it takes too much space. I think i freed about 100 GB... So i would like to merge this soon, so we can test on unstable if anything brakes. I think the chance is quiet good that nothing brakes.

So, can you please add a review again?

**Review by aszlig — APPROVED** [review]

**Comment by davidak** [comment]
I resolved the merge conflict by rebasing. Can we get this merged, so it gets proper testing in unstable before the release?
